android/src/com/mapswithme/maps/downloader/DownloaderAdapter.java

```diff
@@ -610,15 +611,21 @@ void refreshData()
     processData();
   }
 
-  void cancelSearch()
+  void resetSearchResultsMode()
   {
     if (mSearchResultsMode)
+    {
+      mSearchResultsMode = false;
+      mSearchQuery = null;
+      mMyMapsMode = true;
```

Comment:
Нашёл, скачал Перт, mission completed. Имхо - вполне ок. 
Го обсудим с Игорем, @igortomko прокомментируй плиз.

> On Aug 16, 2016, at 11:25 AM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/downloader/DownloaderAdapter.java <https://github.com/mapsme/omim/pull/4053#discussion_r74894103>:
> 
> >    {
> >      if (mSearchResultsMode)
> > +    {
> > +      mSearchResultsMode = false;
> > +      mSearchQuery = null;
> > +      mMyMapsMode = true;
> Я перешёл в поиск карт из корня Австралии, чтобы найти Перт. Нажал отмену, а ты меня кидаешь в список загруженных. По-моему, как-то странно.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/mapsme/omim/pull/4053/files/1924d6e1bfc7e7ad280c569c64db36a887b699af#r74894103>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKTRHbojn-GJ_RzspYbFce1gvkwwoTXks5qgXQKgaJpZM4JjqPO>.
>